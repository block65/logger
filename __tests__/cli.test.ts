import { beforeEach, describe, expect, jest, test } from '@jest/globals';

describe('CLI', () => {
  const oldEnv = process.env;
  beforeEach(() => {
    jest.resetModules();
    process.env = { ...oldEnv };
  });

  test('pretty logger no-color', async () => {
    process.env.NO_COLOR = 'true';
    const { cliLoggerWithWaitableMockWatchOnce } = await import('./helpers.js');

    const [logger, callback] = cliLoggerWithWaitableMockWatchOnce();
    logger.error(new Error('hallo'));
    logger.info(new Error('hello'));
    logger.warn(new Error('halo'));
    logger.debug(new Error('gday'));
    logger.trace(new Error('nihao'));
    await expect(callback.waitUntilCalledTimes(1)).resolves.toMatchSnapshot();
  });

  test('pretty logger force-color', async () => {
    const { cliLoggerWithWaitableMockWatchOnce } = await import('./helpers.js');

    const [logger, callback] = cliLoggerWithWaitableMockWatchOnce({
      color: true,
    });
    logger.error(new Error('hallo'));
    logger.info(new Error('hello'));
    logger.warn(new Error('halo'));
    logger.debug(new Error('gday'));
    logger.trace(new Error('nihao'));
    await expect(callback.waitUntilCalledTimes(1)).resolves.toMatchSnapshot();
  });
});
