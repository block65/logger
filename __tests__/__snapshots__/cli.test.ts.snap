// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI pretty logger force-color 1`] = `
Array [
  Array [
    "2009-02-13T23:31:30.000Z ERROR   hallo { err:
   { message: 'hallo',
     name: 'Error',
     stack:
      'Error: hallo\\\\n    at Object.<anonymous> (<rootDir>/__tests__/cli.test.ts:30:18)' } }

",
  ],
]
`;

exports[`CLI pretty logger no-color 1`] = `
Array [
  Array [
    "2009-02-13T23:31:30.000Z ERROR   hallo { err:
   { message: 'hallo',
     name: 'Error',
     stack:
      'Error: hallo\\\\n    at Object.<anonymous> (<rootDir>/__tests__/cli.test.ts:15:18)\\\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)' } }

",
  ],
]
`;
